L1Trigger/L1TTrackMatch/interface/L1TkMuCorrDynamicWindows.h

```diff
@@ -0,0 +1,154 @@
+
+class L1TkMuCorrDynamicWindows {
+public:
+  typedef TTTrack<Ref_Phase2TrackerDigi_> L1TTTrackType;
+  typedef std::vector<L1TTTrackType> L1TTTrackCollectionType;
+
+  L1TkMuCorrDynamicWindows(std::vector<double>& bounds, TFile* fIn_theta, TFile* fIn_phi);
+  L1TkMuCorrDynamicWindows(
+      std::vector<double>& bounds, TFile* fIn_theta, TFile* fIn_phi, TFile* fIn_theta_S1, TFile* fIn_phi_S1);
+  ~L1TkMuCorrDynamicWindows(){};
+  // void test(double eta, double pt);
+  // std::vector<int> find_match(const EMTFTrackCollection& l1mus, const L1TTTrackCollectionType& l1tks l1trks, std::vector<int>* narbitrated = nullptr); // gives a vector with the idxs of muons for each L1TTT
+  std::vector<int> find_match(
+      const EMTFTrackCollection& l1mus,
+      const L1TTTrackCollectionType& l1trks);  // gives a vector with the idxs of muons for each L1TTT
+  std::vector<int> find_match_stub(
+      const EMTFHitCollection& l1mus,
+      const L1TTTrackCollectionType& l1trks,
+      const int& station,
+      bool requireBX0 = true);  // gives a vector with the idxs of muon stubs from station "station" for each L1TTT
+
+  // ------------------------------
+  static std::vector<double> prepare_corr_bounds(string fname, string hname);
+
+  void set_safety_factor(float sf_l, float sf_h) {
+    safety_factor_l_ = sf_l;
+    safety_factor_h_ = sf_h;
+    // std::cout << "L1TkMuCorrDynamicWindows : safety factor LOW is " << safety_factor_l_ << std::endl;
+    // std::cout << "L1TkMuCorrDynamicWindows : safety factor HIGH is " << safety_factor_h_ << std::endl;
+  }
+  void set_sf_initialrelax(float sf_l, float sf_h) {
+    initial_sf_l_ = sf_l;
+    initial_sf_h_ = sf_h;
+    // std::cout << "L1TkMuCorrDynamicWindows : initial relax safety factor LOW is " << initial_sf_l_ << std::endl;
+    // std::cout << "L1TkMuCorrDynamicWindows : initial relax safety factor HIGH is " << initial_sf_h_ << std::endl;
+  }
+  void set_relaxation_pattern(float pt_start, float pt_end) {
+    pt_start_ = pt_start;
+    pt_end_ = pt_end;
+    // std::cout << "L1TkMuCorrDynamicWindows : set relaxing from " << pt_start_ << " to " << pt_end_ << std::endl;
+  }
+  void set_safety_factor(float sf) { set_safety_factor(sf, sf); }
+  void set_sf_initialrelax(float sf) { set_sf_initialrelax(sf, sf); }
+  void set_do_relax_factor(bool val) {
+    do_relax_factor_ = val;
+    // std::cout << "L1TkMuCorrDynamicWindows : set do_relax to " << std::boolalpha << do_relax_factor_ << std::noboolalpha << std::endl;
+  }
+
+  void set_do_trk_qual_presel(bool val) { track_qual_presel_ = val; }
+
+  // setters for trk
+  void set_n_trk_par(int val) { nTrkPars_ = val; }
+  void set_min_trk_p(float val) { min_trk_p_ = val; }
+  void set_max_trk_aeta(float val) { max_trk_aeta_ = val; }
+  void set_max_trk_chi2(float val) { max_trk_chi2_ = val; }
+  void set_min_trk_nstubs(int val) { min_trk_nstubs_ = val; }
+
+  // getters for trk
+  int n_trk_par() { return nTrkPars_; }
```

Comment:
check if all getters and helper functions can be const functions

Reply:
done